File: class5/Reverse.java
import java.util.*;

public class Reverse {

    public static void swap(int[] arr, int left, int right) {
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }

    public static void reverse(int[] arr) {
        int left = 0;
        int right = arr.length - 1;

        while(left < right) {

            swap (arr, left, right);
            
            left++;
            right--;
        }
    }

    public static void print(int[] arr) {
        for(int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();

    }
    public static void main(String[] args) {
        int[] arr = {10, 20 , 30 , 40, 50, 60};
        reverse(arr);
        print(arr);
        }
}

File: class6/reverse.cpp
#include<bits/stdc++.h>
using namespace std;

void swap(char* a, char* b) {
    char temp = *a;
    *a = *b;
    *b = temp; 
}

void reverse(char* arr, int l, int r) {
    while(l < r) {
        swap(arr[l], arr[r]);
        l++;
        r--;
    }
}

void rotate(char* arr, int size, int k) {
    k %= size;
    k = (k < 0)? size + k: k;
    reverse(arr, 0, size - k - 1);
    reverse(arr, size - k, size - 1);
    reverse(arr, 0 , size - 1);
}

void display(char* arr, int size) {
    for(int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main(int argc, char** argv) {
    char arr[] = {'a', 'b', 'c', 'd' , 'e', 'f', 'g'};
    int rot;
    cout << "Enter the rotation\n";
    cin >> rot;
    int size = (sizeof(arr) / sizeof(arr[0]));
    rotate(arr, size, rot);
    display(arr, size);
}